api/resources/resourcesRouter.js

```diff
@@ -184,22 +190,22 @@ router.get('/', authRequired, async (req, res, next) => {
  *                message:
  *                  type: string
  *                  description: Error message returned by the API
- *                  example: 'Resource not found, check the ID'
+ *                  example: 'Resource with ID 1 not found!'
  */
-router.get('/:resource_id', authRequired, (req, res) => {
-  const id = req.params.resource_id;
-  Resources.findByResourceId(id)
-    .then((resource) => {
-      if (resource) {
-        res.status(200).json(resource);
-      } else {
-        res.status(404).json({ error: 'Resource not found, check the ID' });
-      }
-    })
-    .catch((err) => {
-      res.status(500).json({ error: err.message });
-    });
-});
+
+router.get(
+  '/:resource_id',
+  authRequired,
+  checkResourceIdExists,
+  (req, res, next) => {
+    try {
+      const resource = req._resource;
+      return res.status(200).json(resource);
+    } catch (err) {
+      return next(err);
```

Comment:
Sorry, just curious.. why did you decide to return next(err) here? Would next(err) have been sufficient?

Reply:
Interesting find. If thats the case then the way you're returning next calls is more secure. Keep it like this, it looks clean:)